framework/decode/dx12_json_consumer_base.cpp

```diff
@@ -238,5 +239,63 @@ void Dx12JsonConsumerBase::Process_ExeFileInfo(const util::filepath::FileInfo& i
     FieldToJson(jdata["info_record"], info_record, json_options);
 }
 
+void Dx12JsonConsumerBase::Process_ID3D12Device_CheckFeatureSupport(format::HandleId object_id,
+                                                                    HRESULT          original_result,
+                                                                    D3D12_FEATURE    feature,
+                                                                    const void*      capture_feature_data,
+                                                                    void*            replay_feature_data,
+                                                                    UINT             feature_data_size)
+{
+    using namespace gfxrecon::util;
+    ApiCallInfo call_info;
+    call_info.index     = GetCurrentBlockIndex();
+    call_info.thread_id = 31415926535;
+
+    nlohmann::ordered_json& method =
+        writer_->WriteApiCallStart(call_info, "ID3D12Device", object_id, "CheckFeatureSupport");
+    const JsonOptions& options = writer_->GetOptions();
+    HresultToJson(method[format::kNameReturn], original_result, options);
+    nlohmann::ordered_json& args = method[format::kNameArgs];
+    {
+        FieldToJson(args["Feature"], feature, options);
+        FieldToJson(args["pFeatureSupportData"], nullptr, options);
```

Comment:
This whole processing function is basically a placeholder. Maybe I should just not output that field at all rather than outputting a null to it?

This function needs a lot of work. See how different the actual API signature is to what we are supposed to process:

```cpp
// D3D12:
HRESULT CheckFeatureSupport(
            D3D12_FEATURE Feature,
  [in, out] void          *pFeatureSupportData,
            UINT          FeatureSupportDataSize
);

// GFXRECON:
void Dx12JsonConsumerBase::Process_ID3D12Device_CheckFeatureSupport(format::HandleId object_id,
                                                                    HRESULT          original_result,
                                                                    D3D12_FEATURE    feature,
                                                                    const void*      capture_feature_data,
                                                                    void*            replay_feature_data,
                                                                    UINT             feature_data_size)
```

So what we need to do is probably step through each void pointer in turn, interpreting type fields, extracting structs, converting those structs, in a loop.
See: https://learn.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-checkfeaturesupport

Reply:
This issue now mentions the functions: https://github.com/LunarG/gfxreconstruct/issues/1380